controllers/apps/components/util/component_switchover_util.go

```diff
@@ -0,0 +1,628 @@
+				Namespace: cluster.Namespace,
+				Name:      jobName,
+				Labels:    getSwitchoverCmdJobLabel(cluster.Name, component.Name),
+			},
+			Spec: batchv1.JobSpec{
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: cluster.Namespace,
+						Name:      jobName,
+					},
+					Spec: corev1.PodSpec{
+						RestartPolicy: corev1.RestartPolicyNever,
+						Containers: []corev1.Container{
+							{
+								Name:            constant.KBSwitchoverJobContainerName,
+								Image:           switchoverSpec.Image,
+								ImagePullPolicy: corev1.PullIfNotPresent,
+								Command:         switchoverAction.Command,
+								Args:            switchoverAction.Args,
+								Env:             switchoverEnvs,
+							},
+						},
+					},
+				},
+			},
+		}
+		if len(cluster.Spec.Tolerations) > 0 {
+			job.Spec.Template.Spec.Tolerations = cluster.Spec.Tolerations
+		}
+		return job, nil
+	}
+
+	switchoverEnvs, err := buildSwitchoverEnvs(ctx, cli, cluster, component)
+	if err != nil {
+		return nil, err
+	}
+	job, err := renderJob(component.SwitchoverSpec, switchoverEnvs)
+	if err != nil {
+		return nil, err
+	}
+	return job, nil
+}
+
+// GetJobWithLabels gets the job list with the specified labels.
+func GetJobWithLabels(ctx context.Context,
+	cli client.Client,
+	cluster *appsv1alpha1.Cluster,
+	matchLabels client.MatchingLabels) ([]batchv1.Job, error) {
+	jobList := &batchv1.JobList{}
+	if err := cli.List(ctx, jobList, client.InNamespace(cluster.Namespace), matchLabels); err != nil {
+		return nil, err
+	}
+	return jobList.Items, nil
+}
+
+// CleanJobWithLabels cleans up the job task that execute the switchover commands.
+func CleanJobWithLabels(ctx context.Context,
+	cli client.Client,
+	cluster *appsv1alpha1.Cluster,
+	matchLabels client.MatchingLabels) error {
+	jobList, err := GetJobWithLabels(ctx, cli, cluster, matchLabels)
+	if err != nil {
+		return err
+	}
+	for _, job := range jobList {
+		var ttl = int32(constant.KBJobTTLSecondsAfterFinished)
+		patch := client.MergeFrom(job.DeepCopy())
+		job.Spec.TTLSecondsAfterFinished = &ttl
+		if err := cli.Patch(ctx, &job, patch); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CheckJobSucceed checks the result of job execution.
+func CheckJobSucceed(ctx context.Context,
+	cli client.Client,
+	cluster *appsv1alpha1.Cluster,
+	job *batchv1.Job) error {
+	key := types.NamespacedName{Namespace: cluster.Namespace, Name: job.Name}
+	currentJob := batchv1.Job{}
+	exists, err := intctrlutil.CheckResourceExists(ctx, cli, key, &currentJob)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return errors.New("job not exist, pls check.")
+	}
+	jobStatusConditions := currentJob.Status.Conditions
+	if len(jobStatusConditions) > 0 {
+		switch jobStatusConditions[0].Type {
+		case batchv1.JobComplete:
+			return nil
+		case batchv1.JobFailed:
+			return errors.New("job failed, pls check.")
+		default:
+			return errors.New("job unfinished.")
```

Comment:
should not return error for running job?

Reply:
> should not return error for running job?

requeue when job unfinished